src/Val.cc

```diff
@@ -1925,6 +1943,66 @@ ListVal* TableVal::RecoverIndex(const HashKey* k) const
 	return table_hash->RecoverVals(k);
 	}
 
+void TableVal::CallChangeFunc(const Val* index, Val* old_value, OnChangeType tpe)
+	{
+	if ( ! change_func || ! index || in_change_func )
+		return;
+
+	if ( ! table_type->IsSet() && ! old_value )
+		return;
+
+	try
+		{
+		Val* thefunc = change_func->Eval(0);
```

Comment:
```suggestion
		IntrusivePtr<Val> thefunc{change_func->Eval(nullptr), false};
```

Then can remove two `Unref(thefunc)` calls below.  i.e. prevents leaking it in case of interpreter exception.  The `val_list` would still leak at the moment, but I imagine a later overhaul that moves more things to `IntrusivePtr` will end up fixing it so probably wouldn't worry about it (or just make a note).